beacon-chain/blockchain/process_block.go

```diff
@@ -251,19 +211,19 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.SignedBeaconBlo
 	}()
 
 	// Save justified check point to db.
-	if postState.CurrentJustifiedCheckpoint().Epoch > currJustifiedEpoch {
+	if justified.Epoch > currJustifiedEpoch {
```

Comment:
this seems like a big change, instead of using the post state's value we will instead use it from our forkchoice service. This doesn't seems to have anything to do with store, so why are we changing it ?

Reply:
Because we want to guarantee to have DB to agree with forkchoice, the previous behavior was wrong, there are checks that one needs to make before saving a justified checkpoint, forkchoice does them, but the post-state does not.